FinniversKit/Sources/Fullscreen/FavoriteSoldView/FavoriteSoldView.swift

```diff
@@ -0,0 +1,265 @@
+    }
+
+    // MARK: - Overrides
+
+    public override func layoutSubviews() {
+        super.layoutSubviews()
+
+        if didSetupView == false {
+            setup()
+            didSetupView = true
+        } else if !boundsForCurrentSubviewSetup.equalTo(bounds) {
+            setupFrames()
+        }
+    }
+
+    // MARK: - Setup
+
+    private func setup() {
+        addSubview(adsGridView)
+        adsGridView.collectionView.addSubview(adsRetryView)
+
+        headerView.addSubview(stackView)
+        headerView.addSubview(imageContentView)
+        headerView.addSubview(similarAdsTitleLabel)
+
+        imageContentView.addSubview(imageView)
+        imageView.fillInSuperview()
+
+        stackView.addArrangedSubview(ribbonView)
+        stackView.addArrangedSubview(titleLabel)
+        stackView.addArrangedSubview(bodyLabel)
+
+        for subview in stackView.arrangedSubviews {
+            subview.setContentHuggingPriority(.defaultHigh, for: .vertical)
+        }
+        let spaceFiller = UIView(withAutoLayout: true)
+        spaceFiller.setContentHuggingPriority(.defaultLow, for: .vertical)
+        stackView.addArrangedSubview(spaceFiller)
+
+        stackView.setCustomSpacing(FavoriteSoldView.titleTopSpacing, after: ribbonView)
+        stackView.setCustomSpacing(FavoriteSoldView.bodyTopSpacing, after: titleLabel)
+
+        NSLayoutConstraint.activate([
+            imageContentView.topAnchor.constraint(equalTo: headerView.topAnchor, constant: FavoriteSoldView.margins),
+            imageContentView.widthAnchor.constraint(equalToConstant: FavoriteSoldView.imageWidth),
+            imageContentView.heightAnchor.constraint(equalTo: imageContentView.widthAnchor),
+            imageContentView.leadingAnchor.constraint(equalTo: headerView.leadingAnchor, constant: FavoriteSoldView.margins),
+
+            stackView.leadingAnchor.constraint(equalTo: imageContentView.trailingAnchor, constant: FavoriteSoldView.margins),
+            stackView.topAnchor.constraint(equalTo: imageContentView.topAnchor),
+            stackView.trailingAnchor.constraint(equalTo: headerView.trailingAnchor, constant: -FavoriteSoldView.margins),
+            stackView.bottomAnchor.constraint(greaterThanOrEqualTo: imageContentView.bottomAnchor),
+
+            similarAdsTitleLabel.topAnchor.constraint(equalTo: stackView.bottomAnchor, constant: FavoriteSoldView.similarAdsTitleTopSpacing),
+            similarAdsTitleLabel.leadingAnchor.constraint(equalTo: headerView.leadingAnchor, constant: FavoriteSoldView.margins),
+            similarAdsTitleLabel.trailingAnchor.constraint(equalTo: headerView.trailingAnchor, constant: -FavoriteSoldView.margins),
+        ])
+
+        adsGridView.fillInSuperview()
+        adsGridView.headerView = headerView
+        setupFrames()
+    }
+
+    private func setupFrames() {
+        let stackViewWidth = frame.size.width - FavoriteSoldView.imageWidth - 3 * FavoriteSoldView.margins
```

Comment:
Have you tried using `headerView.systemLayoutSizeFitting` instead? You can use that to calculate its size using its current constraints. Use it like this:
```
let targetSize = CGSize(width: frame.width, height: 0)
let size = headerView.systemLayoutSizeFitting(
    targetSize, 
    withHorizontalFittingPriority: .required,
    verticalFittingPriority: .defaultLow
)
```
That should in theory give you the smallest height given the width of the frame

Reply:
I tried now, it doesn't seem to work. I get height 0. Maybe that's because `headerView` is a `UICollectionView.elementKindSectionHeader` of a collection view. I'm not sure if it's supposed to work in that case?

Reply:
I think the problem is that you haven't constrained the `similarAdsTitleLabel.bottomAnchor` to `headerView.bottomAnchor`.

Reply:
Yes, it worked! Thanks, that's much nicer 😄